eng/common/pipelines/templates/validate-filename.yml

```diff
@@ -0,0 +1,16 @@
+parameters:
+  WorkingDirectory: $(System.DefaultWorkingDirectory)
+steps:
+- pwsh: |
+    $differByCaseFiles = git ls-files | Group-Object | Where-Object { $_.Count -gt 1 } | ForEach-Object { $_.Group} 
+
+    if ($differByCaseFiles) 
+    {
+      Write-Error "Do NOT name the files which only differ in case. Please check and rename the following files:"
+      $differByCaseFiles | Write-Error
```

Comment:
I think you might need to dump these to Write-Host and before the Write-Error line.